union_pooling/union_pooling/nupic_integration/custom_region/UPRegion.py

```diff
@@ -0,0 +1,349 @@
+  #  computed automatically
+  ourArgNames += [
+    "inputDimensions",
+  ]
+  for argTuple in argTuples[:]:
+    if argTuple[0] in ourArgNames:
+      argTuples.remove(argTuple)
+
+  # Build the dictionary of arguments
+  if self:
+    for argTuple in argTuples:
+      argName = argTuple[0]
+      if argName in kwargs:
+        # Argument was provided
+        argValue = kwargs.pop(argName)
+      else:
+        # Argument was not provided; use the default value if there is one, and
+        #  raise an exception otherwise
+        if len(argTuple) == 2:
+          # No default value
+          raise TypeError("Must provide value for '%s'" % argName)
+        argValue = argTuple[2]
+      # Set as an instance variable if "self" was passed in
+      setattr(self, argName, argValue)
+
+  return argTuples
+
+
+def _getAdditionalSpecs():
+  """Build the additional specs in three groups (for the inspector)
+
+  Use the type of the default argument to set the Spec type, defaulting
+  to "Byte" for None and complex types
+
+  Determines the union parameters based on the selected implementation.
+  It defaults to UnionPooler.
+  """
+  typeNames = {int: "UInt32", float: "Real32", str: "Byte", bool: "bool", tuple: "tuple"}
+
+  def getArgType(arg):
+    t = typeNames.get(type(arg), "Byte")
+    count = 0 if t == "Byte" else 1
+    if t == "tuple":
+      t = typeNames.get(type(arg[0]), "Byte")
+      count = len(arg)
+    if t == "bool":
+      t = "UInt32"
+    return (t, count)
+
+  def getConstraints(arg):
+    t = typeNames.get(type(arg), "Byte")
+    if t == "Byte":
+      return "multiple"
+    elif t == "bool":
+      return "bool"
+    else:
+      return ""
+
+  # Get arguments from union pooler constructors, figure out types of
+  # variables and populate unionSpec.
+  uArgTuples = _buildArgs()
+  unionSpec = {}
+  for argTuple in uArgTuples:
+    d = dict(
+      description=argTuple[1],
+      accessMode="ReadWrite",
+      dataType=getArgType(argTuple[2])[0],
+      count=getArgType(argTuple[2])[1],
+      constraints=getConstraints(argTuple[2]))
+    unionSpec[argTuple[0]] = d
+
+  # Add special parameters that weren't handled automatically
+  # Union pooler parameters only
+  unionSpec.update(dict(
+    columnCount=dict(
+      description="Total number of columns (coincidences).",
+      accessMode="ReadWrite",
+      dataType="UInt32",
+      count=1,
+      constraints=""),
+
+    inputWidth=dict(
+      description="Size of inputs to the UP.",
+      accessMode="ReadWrite",
+      dataType="UInt32",
+      count=1,
+      constraints=""),
+    ))
+
+
+  # The last group is for parameters that aren't specific to union pooler
+  otherSpec = dict(
+    learningMode=dict(
+      description="1 if the node is learning (default 1).",
+      accessMode="ReadWrite",
+      dataType="bool",
+      count=1,
+      constraints="bool"),
+  )
+
+  return unionSpec, otherSpec
+
+
+
+class UPRegion(PyRegion):
```

Comment:
`PoolingRegion` would be a better name in my opion, especially since we can probably use it for other pooling implementations